src/components/EvolutionTab.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import { fetchPokemon } from "../util/fetchPokemonData";
+
+const EvolutionTab = ({ evolutionChainUrl }) => {
+  const [evolutionNamesObjects, setEvolutionNamesObjects] = useState([]);
+  const [evolutionNameStrings, setEvolutionNameStrings] = useState([]);
+  const [spriteUrls, setSpriteUrls] = useState([]);
+
+  let evolutionNamesList = [];
+  let chain = {};
+
+  // Function to fetch chain object, save to state hook
+  const fetchChain = async () => {
+    try {
+      const chainData = await fetch(evolutionChainUrl)
+        .then((response) => response.json())
+        .then((data) => {
+          chain = data.chain;
+        });
+      evolutionNamesList = getEvolutionNames(chain);
+      setEvolutionNamesObjects(evolutionNamesList);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  // search through each pokemon object and get all names of evolution into array
+  const getEvolutionNames = (chainObj) => {
+    let evoChain = [];
+    let evoData = chainObj;
+
+    if (chainObj) {
+      do {
+        let numberOfEvolutions = evoData["evolves_to"].length;
+        evoChain.push({
+          species_name: evoData.species.name,
+        });
+
+        if (numberOfEvolutions > 1) {
+          for (let i = 1; i < numberOfEvolutions; i++) {
+            evoChain.push({
+              species_name: evoData.evolves_to[i].species.name,
+            });
+          }
+        }
+
+        evoData = evoData["evolves_to"][0];
+      } while (!!evoData && evoData.hasOwnProperty("evolves_to"));
+      return evoChain;
```

Comment:
Interesting way of getting all the Pokémon's evolutions

Reply:
This evolution stuff looks intense! Nice work figuring it out.

Reply:
stack overflow ftw